tests/subsys/suit/unit/suit_decrypt_filter/src/main.c

```diff
@@ -0,0 +1,545 @@
+			"Incorrect error code when calling filter write interface");
+	zassert_equal(fixture->ram_sink_write_call_cnt, 2,
+			"Incorrect number of ram_sink write function calls");
+}
+
+ZTEST_F(suit_decrypt_filter_tests, test_flush_verify_fail)
+{
+	struct suit_encryption_info enc_info = ENC_INFO_DEFAULT_INIT;
+
+	suit_plat_decode_key_id_fake.custom_fake = custom_suit_plat_decode_key_id;
+	suit_plat_err_t err = suit_decrypt_filter_get(&fixture->dec_sink, &enc_info,
+							&sample_class_id, &fixture->ram_sink);
+
+	zassert_equal(err, SUIT_PLAT_SUCCESS,
+			"Incorrect error code when getting decrypt filter");
+
+	psa_aead_update_fake.custom_fake = custom_psa_aead_update;
+	fixture->aead_update_status = PSA_SUCCESS;
+	fixture->decrypted_output_length = 100; /* Anything greater than 0. */
+
+	err = fixture->dec_sink.write(fixture->dec_sink.ctx, ciphertext_direct,
+								sizeof(ciphertext_direct));
+
+	zassert_equal(err, SUIT_PLAT_SUCCESS,
+			"Incorrect error code when calling filter write interface");
+
+	psa_aead_verify_fake.custom_fake = custom_psa_aead_verify;
+	fixture->aead_verify_status = PSA_ERROR_GENERIC_ERROR;
+
+	err = fixture->dec_sink.flush(fixture->dec_sink.ctx);
+
+	zassert_equal(err, SUIT_PLAT_ERR_AUTHENTICATION,
+			"Incorrect error code when calling filter flush interface");
+	zassert_equal(fixture->ram_sink_erase_call_cnt, 1,
+			"Incorrect number of ram_sink write function calls");
+	zassert_equal(psa_aead_abort_fake.call_count, 1,
+			"Invalid number of calls to psa_aead_abort");
+}
+
+ZTEST_F(suit_decrypt_filter_tests, test_flush_happy_path)
+{
+	struct suit_encryption_info enc_info = ENC_INFO_DEFAULT_INIT;
+
+	suit_plat_decode_key_id_fake.custom_fake = custom_suit_plat_decode_key_id;
+	suit_plat_err_t err = suit_decrypt_filter_get(&fixture->dec_sink, &enc_info,
+							&sample_class_id, &fixture->ram_sink);
+
+	zassert_equal(err, SUIT_PLAT_SUCCESS,
+			"Incorrect error code when getting decrypt filter");
+
+	psa_aead_update_fake.custom_fake = custom_psa_aead_update;
+	fixture->aead_update_status = PSA_SUCCESS;
+	fixture->decrypted_output_length = 100; /* Anything greater than 0. */
+
+	err = fixture->dec_sink.write(fixture->dec_sink.ctx, ciphertext_direct,
+								sizeof(ciphertext_direct));
+
+	zassert_equal(err, SUIT_PLAT_SUCCESS,
+			"Incorrect error code when calling filter write interface");
+
+	psa_aead_verify_fake.custom_fake = custom_psa_aead_verify;
+	fixture->aead_verify_status = PSA_SUCCESS;
+	fixture->decrypted_output_length = 100; /* Anything greater than 0. */
+
+	err = fixture->dec_sink.flush(fixture->dec_sink.ctx);
+
+	zassert_equal(err, SUIT_PLAT_SUCCESS,
+			"Incorrect error code when calling filter flush interface");
+	zassert_equal(fixture->ram_sink_write_call_cnt, 2,
+			"Incorrect number of ram_sink write function calls");
+	zassert_equal(psa_aead_abort_fake.call_count, 0,
+			"Invalid number of calls to psa_aead_abort");
+}
+
+ZTEST_F(suit_decrypt_filter_tests, test_flush_ram_write_fail)
```

Comment:
Rather name it `test_flush_out_sink_write_fail` ? In general, the decrypt filter does not write to RAM